### ISSUE TITLE
Diagnose Odd Behaviour for solo Xenos in Cheyenne

### ISSUE THREAD

**Issue by IrateRedKite** [better-modernized-combat/bmod-client#75]
Solo Xenos in Cheyenne at the moment seems to be very reluctant to engage in MP. This needs investigating as Cheyenne is supposed to be a reliable source of 1v1s for new players.

**Comment by FJonske** [comment]
This is related to the [JobBlock] of the respective jobs of spawns in Cheyenne and a quirk with the current equipment. The jobs of enemies in the fields are one of assault_leader_job, assault_job and scout_job. The former two have the key 'scene_threshold_toughness' set to 'hard' instead of 'hardest' like the scout_job. This makes them unable to/unwilling to engage based on perceived player toughness. The actual fleeing behavior is governed by a separate key, called 'flee_scene_threat_style', which is 'hardest' for all three right now, leading to them seeing the player as too tough to engage alone, but not tough enough to outright run from (a behavior we can exploit in the future, with enough fine-tuning). Notably, they do NOT fight back, only evade or hover when in this inbetween state.

The reason that a single player already exceeds the 'hard' threshold with their toughness comes from the player equipment. Specifically, I think, the shields, which have immensely high values right now.

A quick solution is to set everything to 'hardest' for now and just have the Xenos be super bloodthirsty and always engage. In the future, we want to look at how we set up toughness, what a realistic player toughness would be for a new player and a fully decked out player, and where the corresponding threshold should be. I am not sure where they are saved, I dimly recall hearing about 'ptough.ini', though?

Leaving this open until I commit JobBlocks with all 'hardest' settings. I think finetuning this stuff is out of scope for 0.1

**Comment by Beaglerush** [comment]
"In the future, we want to look at how we set up toughness, what a realistic player toughness would be for a new player and a fully decked out player, and where the corresponding threshold should be."

I haven't yet started to think about how I want to properly design around this system, but one thing I'll immediately say: I would first be considering using this to rate different classes of threats to the AI rather than using it for actual equipment-based judgement. So for instance we would make smart AI unwilling to engage Heavy Fighters alone, unwilling to engage VHFs without 3 or more, and unwilling to engage Freighters without a big squad, preferably with bomber equipment. The last point is why it would matter if this works both ways, if the AI considers their own toughness and compares it to the player - we would want to give freighters and frigates big toughness, but then also give big toughness gains to bomber equipment like torpedoes and anti-cap assault guns, thereby letting the AI judge if they have the "anti-tank" hardware and the numbers needed to take on a player in a turreted gunboat or not, and when to cut their losses and flee

**Comment by FJonske** [comment]
A smart AI shouldn't avoid 1v1 engagements, should it? That seems overly defensive.

I don't know how fine-grained we can be with this system, because I don't know whether the hardcoded limits of "hardest" and "hard" are extensible. Might be that we can't even change the levels themselves, just their values, so that we can only really differentiate in very coarse manner.

**Comment by Beaglerush** [comment]
Less a smart AI and more a smart enemy pilot in narrative would prefer to wait for an unfair fight than take the 1v1, depending on faction. Something to play with per context anyway. I'm interested to hear more about your testing.

**Comment by FJonske** [comment]
Sure, a smart enemy would take unfair fights. But in consequence, that leads to players chasing a wholly-defensive or fleeing AI around until reinforcements show up or theyre dead. If I were a player, that would annoy me. If the VHF threshold was higher, even more so. Imagine you can't even fight a 1v2 anymore, because your opponents are constantly running off. I'd strongly limit which factions actually do this.

**Comment by Beaglerush** [comment]
I wouldn't worry too much about it yet; I promise you I'll be mindful of that in implementation, practice and testing. The goal would be to avoid braindead AI suicide into unwinnable fights, rather than making similarly unfun situations on the opposite end with constantly scaredy-cat AI.

**Comment by Beaglerush** [comment]
I spent some time working with this, this morning, and behaviours are happening pretty much as expected now! See my latest commits, I've updated the Cheyenne field encounters to use new specific JobBlocks that have the Xenos engaging consistently and then fleeing when damaged, all as intended.

When I first learned to work with the AI in pilots_population back years ago, I'd yet to learn what the encounters file was or how to use it, and so I never put together that the job called in the encounters is the same one as the [JobBlock] entries created in pilots_population. Because the summarized pilot blocks had their -own- Jobs defined, I assumed they would respect that as their behaviour, not realising that the encounter jobs were overwriting the jobs I was assigning to their pilot blocks!
Now that you've figured this out, everything suddenly makes more sense as the connection has clicked into place. Thanks hugely for figuring this out Katz, I'm only on day one of experimenting beyond the basics but this has instantly given us way more control and potential for our AI's ability.

I've got control over this now so no need to commit anything to the pilots pop as you planned, leave it to me. I think we can call this closed thanks to Katz.
